### ISSUE TITLE
Remove the need to hold "shift" key to move a crop.

### ISSUE THREAD

**Issue by maltmoose** [Beep6581/RawTherapee#5076]
Windows 10
Rawtherapee 5.5 rc1

After you have created a selection in the crop tool, if you want to then move the selected crop, you have to hold the "shift key" to move the crop.
This is an unnecessary step and requires 2 hands to move a crop. (one hand for shift and one had on mouse.)
Removing the need to hold the shift key will improve cropping work flow a lot.

**How it should work:-
It should work like the color picker where you just "left click and drag"**

Thank you

**Comment by Thanatomanic** [comment]
Thank you for your input. There are several 'issues' and opinions regarding the crop tool. Just to indicate: none will be addressed for 5.5, but we may discuss changes for 5.6.

**Comment by TooWaBoo** [comment]
@maltmoose 
> left click and drag

... and how would you drag the image around? The left click + drag is already used.

**Comment by ff2000** [comment]
> @maltmoose
> 
> > left click and drag
> 
> ... and how would you drag the image around? The left click + drag is already used.

e.g. create a "permanent" crop tool that doesn't fall back to "hand tool" as soon as you release the mouse button after you pulled your crop rect.
Or make the crop itself permanent as it seems lightroom implements it. Currently it is just a selection that darkens (or overpaints) the regions outside the crop rectangle. I saw this requested earlier.

**Comment by TooWaBoo** [comment]
Everyone has a differrent workflow.
I draw the crop rectangle, finetune it with the crop border and the shift drag feature helps me to prevent to move the crop by accident. For me it's perfect how it works now.

**Comment by ff2000** [comment]
@TooWaBoo you obviously don't have cats. They usually want to sit in your arm when you are working on your images. In such cases I am happy if I can navigate my desktop, applications, ... with just the mouse ;) (Currently I fall back to dragging the borders in such cases, but I can see that dragging the whole selection is easier for others.)

**Comment by TooWaBoo** [comment]
@ff2000 
Teach your cat to hold the shift key. 😸

**Comment by maltmoose** [comment]
> 
> 
> @maltmoose
> 
> > left click and drag
> 
> ... and how would you drag the image around? The left click + drag is already used.

Just needs to be implemented like most other software like affinity, dxo, capture one, faststone.

When a crop selection is made you just left click to drag the crop just made, simples.

**Comment by heckflosse** [comment]
>> with just the mouse

> Teach your cat to hold the shift key.

Mission impossible. Cat will take care of the mouse, not the shift key ;-)

**Comment by TooWaBoo** [comment]
@maltmoose 
> When a crop selection is made you just left click to drag the crop just made, simples.

And what do I have to do when I want to move the image and not the crop after I've set the crop?

**Comment by Thanatomanic** [comment]
A 'Lock crop' or 'Commit crop' or 'Accept crop' or whatever could solve so many issues. If we would not allow to modify the crop if the crop tool is not active, I think we would have a good compromise for either workflow.

**Comment by TooWaBoo** [comment]
@Thanatomanic
This means we need another tool button - right?

**Comment by Thanatomanic** [comment]
@toowaboo not necessarily. The 'c' hotkey enables the 'Select crop' option. If the 'Select crop' in the Crop tool becomes a toggle button, it would be solved imo.

**Comment by maltmoose** [comment]
> 
> 
> @maltmoose
> 
> > When a crop selection is made you just left click to drag the crop just made, simples.
> 
> And what do I have to do when I want to move the image and not the crop after I've set the crop?

I would suggest implementing clicking on the hand tool as the comit crop action.

From a workflow perspective i cant see anyone wanting to try moving the entire image and also try to create a crop selection at the same time. your in the crop tool to crop and an important part of a crop is moving the selection with ease so even your cat can do it.

**Comment by heckflosse** [comment]
@maltmoose 

> From a workflow perspective i cant see anyone wanting to try moving the entire image and also try to create a crop selection at the same time. your in the crop tool to crop and an important part of a crop is moving the selection with ease so even your cat can do it.

I really tried to understand what you wrote. But I can't . Please rephrase in clear words. I have a dog btw...

**Comment by TooWaBoo** [comment]
@maltmoose
> From a workflow perspective i cant see anyone wanting to try moving the entire image...

Me, I set the crop area and zoom in for fine tuning with the crop border. If "Moving Crop" is active, it could be that I move the crop area by accident instead of changing the crop border 'cause I don't see the whole image and in this workflow I need to move the image.
When you work in a zoomed image moving the image should be as easy as possible. If I always have to switch between moving image and crop edit makes all things more complicate.
I often use the double click to zoom in or out. If "Mooving Crop" is active, I have to click very carefully to not move the crop area.

You should look at the whole edit workflow and not only on an isolated function.

**Comment by maltmoose** [comment]
> 
> 
> @maltmoose
> 
> > From a workflow perspective i cant see anyone wanting to try moving the entire image...
> 
> Me, I set the crop area and zoom in for fine tuning with the crop border. If "Moving Crop" is active, it could be that I move the crop area by accident instead of changing the crop border 'cause I don't see the whole image and in this workflow I need to move the image.
> When you work in a zoomed image moving the image should be as easy as possible. If I always have to switch between moving image and crop edit makes all things more complicate.
> I often use the double click to zoom in or out. If "Mooving Crop" is active, I have to click very carefully to not move the crop area.
> 
> You should look at the whole edit workflow and not only on an isolated function.

I have no idea what you are doing but hopefully common sense will prevail.
Darktable whilst not perfect has a better crop method, hopefully can be lifted from that. Theres just no need for a shift key to move a crop ever if the tool is well implemented.

**Comment by heckflosse** [comment]
@maltmoose 
> Darktable whilst not perfect has a better crop method, hopefully can be lifted from that. Theres just no need for a shift key to move a crop ever if the tool is well implemented.

please be more concrete instead of throwing vague sentences

**Comment by Thanatomanic** [comment]
@heckflosse He did try to explain it before.

> I would suggest implementing clicking on the hand tool as the comit crop action.

For what it's worth, I would propose that we end the discussion now, but think hard on what we want to do with the current crop tool in RT's future. We can never please everyone, but how the crop tool is handled seems like a recurring theme, and the preview + crop is a core functionality of any image processing software. So if we have something to improve, we should try to know what we want, and why.
So maybe we should draft a small comparison that outlines RT's behavior w.r.t. the preview window and crop tool, and compare it with other software. Maybe make a user poll on the forum on how people use it, or would want to use it?
My standpoint would be this: if all other software uses the same (or a very similar) way to handle the preview + crop, there must be a very strong argument why we should not adopt the same behavior in RT simply out of UX perspective. Or am I now completely off-base?

I would be willing to work on such a comparison.

**Comment by Hombre57** [comment]
@Thanatomanic I don't see the need for a user poll ; as you suggested, the Crop tool could work like the Lockable Color Picker. When the LCP tool is enabled, you can pan the image as well as add/move the pickers without using the shift key, depending on the hover condition. Once done, you right click to go back to the Hand mode. The pickers are still there, but you can't act on them anymore. You have to enter the LCP mode to edit them again.

The actual workflow for the crop tool is to let the user edit the crop just by pressing the shift key at any time. The nice (but maybe confusing) thing is that both can be implemented at the same time (i.e. using the shift key to edit the crop and entering the Crop mode to edit it w/o shift key).

Everyone pleased ?

**Comment by TooWaBoo** [comment]
@Hombre57 
...but thre is no `crop mode`, it's only a `crop draw mode`.

**Comment by Hombre57** [comment]
@TooWaBoo as of now, yes, but it can be changed easily I think.

**Comment by Kildor** [comment]
I vote for current behaviour.
To really productive workflow user need two hands anyway, there are many hotkeys. Also there could be some predefinded profiles with enabled crop.

It is nice to have ability to switch between two different workflows, but users should be able to switch to old one.

**Comment by Hombre57** [comment]
All in all, it makes sense to set and edit the crop without holding the shift key if the crop button is set, that's something we can do while preserving the old behavior when the crop button is released.

**Comment by marcelloromani** [comment]
Thanks to this bug report, I found out I need to hold SHIFT down to move the crop, where I thought RT didn't have the function! So thanks to the reporter

**Comment by kacnje** [comment]
OMG, how much time did I lose? I did uninstall RawTherapee twice. And today I searched if moving a crop selection is maybe a known bug. And I found that I have to hold SHIFT to move the selection.
